examples/cluster/plot_cluster_comparison.py

```diff
@@ -35,33 +34,46 @@
 from sklearn.preprocessing import StandardScaler
 from itertools import cycle, islice
 
-np.random.seed(0)
 
-# ============
-# Generate datasets. We choose the size big enough to see the scalability
-# of the algorithms, but not too big to avoid too long running times
-# ============
+# %%
+# Generate sample data
+# ---------------------------------
+
+# We choose the size big enough to see the scalability
+# of the algorithms, but not too big to avoid too long running times.
+# Sample data consists of
+# 1. noisy circles 2. noisy moons 3. blobs with varied variance
+# 4. anisotrophic distributions ('stripes') 5. blobs with smaller variance
+# 6. rectangular noise
+
 n_samples = 500
+# 1. make large circle containing smaller circle
```

Comment:
I would remove those inline python comments for each dataset. Instead, I would extend the description above by providing a bullet point, e.g.:

```python
# The sample datasets consist of:
# 1. `noisy_circle`: make a large circle containing a smaller circle
# ...
```

Potentially, we could add an explanation why the dataset is known to be challenging for some clustering methods (when this is the case).